src/options/README.md

```diff
@@ -0,0 +1,45 @@
+# Options Parser - Dev Documentation
+
+## How to add an argument to the parser
+
+### Different types of arguments
+
+If your argument does not take a value please use this syntax:
+```rust
+///Description of the option
+#[arg(short ='<character to use>', long, action = clap::ArgAction::Count)]
+pub <name of the argument>: u8;
+```
+
+If Your argument takes a value please use this syntax instead:
+```rust
+///Description of the option
+#[arg(short ='<character to use>', long)]
+pub <name of the argument>: Option<type>;
+```
+
+Please also change the different [completions](../../completions/) to add your argument too
+
+For any other more complex usage please refer to [Clap documentation on arguments](https://docs.rs/clap/latest/clap/struct.Arg.html#) (please remeber that anything shown in this can be use even tho we using the derive version)
```

Comment:
```suggestion
For any other more complex usage please refer to [Clap documentation on arguments](https://docs.rs/clap/latest/clap/struct.Arg.html#) (please remember that anything shown in this can be use even tho we using the derive version)
```